internal/tool/openssf_scanner.go

```diff
@@ -0,0 +1,546 @@
+	if err := json.Unmarshal(data, &f); err != nil {
+		log.Printf("Warning: Failed to parse OSV file %s: %v", filePath, err)
+		return
+	}
+
+	s.addAffectedPackagesToIndex(f, idxMu)
+}
+
+// addAffectedPackagesToIndex adds affected packages from an OSV file to the index
+func (s *OpenSSFScanner) addAffectedPackagesToIndex(f osvFile, idxMu *sync.Mutex) {
+	for _, aff := range f.Affected {
+		eco := strings.ToLower(aff.Package.Ecosystem)
+		name := strings.ToLower(aff.Package.Name)
+		if eco == "" || name == "" {
+			continue
+		}
+
+		entry := osvShallow{ID: f.ID, Summary: f.Summary, Versions: aff.Versions, Ranges: aff.Ranges}
+		s.addEntryToIndex(eco, name, entry, idxMu)
+	}
+}
+
+// addEntryToIndex adds a single entry to the index with proper locking
+func (s *OpenSSFScanner) addEntryToIndex(eco, name string, entry osvShallow, idxMu *sync.Mutex) {
+	idxMu.Lock()
+	defer idxMu.Unlock()
+
+	m, ok := s.index[eco]
+	if !ok {
+		m = make(map[string][]osvShallow)
+		s.index[eco] = m
+	}
+	m[name] = append(m[name], entry)
+}
+
+// versionMatches checks if a version matches the malicious package criteria
+func (s *OpenSSFScanner) versionMatches(version string, affectedVersions []string, ranges []osvRange) bool {
+	for _, affectedVersion := range affectedVersions {
+		if version == affectedVersion {
+			return true
+		}
+	}
+	for _, versionRange := range ranges {
+		if versionRange.Type == "SEMVER" || versionRange.Type == "ECOSYSTEM" {
+			if s.checkSingleVersionRange(version, versionRange.Events) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (s *OpenSSFScanner) checkSingleVersionRange(version string, events []struct {
+	Introduced string `json:"introduced,omitempty"`
+	Fixed      string `json:"fixed,omitempty"`
+}) bool {
+	for _, event := range events {
+		if s.isVersionInRange(version, event) {
+			return true
+		}
+	}
+	return false
+}
+
+// isVersionInRange checks if a version falls within a specific event range
+func (s *OpenSSFScanner) isVersionInRange(version string, event struct {
+	Introduced string `json:"introduced,omitempty"`
+	Fixed      string `json:"fixed,omitempty"`
+}) bool {
+	if event.Introduced != "" && event.Fixed != "" {
+		return s.isVersionBetween(version, event.Introduced, event.Fixed)
+	}
+	if event.Introduced != "" {
+		return s.isVersionAtLeast(version, event.Introduced)
+	}
+	return false
+}
+
+// isVersionBetween checks if version is >= introduced and < fixed
+func (s *OpenSSFScanner) isVersionBetween(version, introduced, fixed string) bool {
+	return semver.Compare("v"+version, "v"+introduced) >= 0 &&
+		semver.Compare("v"+version, "v"+fixed) < 0
+}
+
+// isVersionAtLeast checks if version is >= introduced
+func (s *OpenSSFScanner) isVersionAtLeast(version, introduced string) bool {
+	return semver.Compare("v"+version, "v"+introduced) >= 0
+}
+
+func (s *OpenSSFScanner) findPackageLineNumber(sourceDir, fileName, pkgName string) int {
+	return fallbackSearchForLineNumber(sourceDir, fileName, pkgName)
```

Comment:
This simply uses the fallback. Please use the same logic in `tool.go` get the line number.